fe/fe-core/src/main/java/com/starrocks/service/FrontendServiceImpl.java

```diff
@@ -2339,7 +2339,12 @@ private static TCreatePartitionResult createPartitionProcess(TCreatePartitionReq
                 LOG.warn("cancel schema change or rollup failed. error: {}", e.getMessage());
             }
 
-            state.getLocalMetastore().addPartitions(db, olapTable.getName(), addPartitionClause);
+            // If a create partition request is from BE or CN, the warehouse information may be lost, we can get it from txn state.
+            ConnectContext ctx = com.starrocks.common.util.Util.getOrCreateConnectContext();
```

Comment:
is this warehouse id right? seems not relate to ingestion context

Reply:
Yes, There is no contxt in FrontendServiceImpl.java.

The reason why I created a new ConnectContext object is that I want to pass warehouse to the methods. But passing it by parameter is not proper.  @HangyuanLiu 

There are similar usages like this
```
                ConnectContext connectContext = new ConnectContext();
                Scope sourceScope = new Scope(RelationId.anonymous(),
                        new RelationFields(table.getBaseSchema().stream().map(col ->
                                        new Field(col.getName(), col.getType(), null, null))
                                .collect(Collectors.toList())));
                Scope outputScope = new Scope(RelationId.anonymous(),
                        new RelationFields(table.getBaseSchema().stream().map(col ->
                                        new Field(col.getName(), col.getType(), new TableName(dbName, table.getName()), null))
                                .collect(Collectors.toList())));
                SelectAnalyzer.RewriteAliasVisitor visitor =
                        new SelectAnalyzer.RewriteAliasVisitor(sourceScope, outputScope,
                                outputExprs, connectContext);

```